opendevin/server/listen.py

```diff
@@ -118,10 +118,18 @@ def read_default_model():
     return config.get_fe_config()
 
 
-@app.get('/api/refresh-files')
-def refresh_files():
-    structure = files.get_folder_structure(Path(str(config.get('WORKSPACE_BASE'))))
-    return structure.to_dict()
+@app.get('/api/list-files')
+def refresh_files(relpath: str = Query(None, description='Relative path from workspace base')):
+    """Refreshes and returns the files and directories from a specified subdirectory or the base directory if no subdirectory is specified, limited to one level deep."""
+    base_path = Path(config.get('WORKSPACE_BASE')).resolve()
+    full_path = (base_path / relpath).resolve() if relpath is not None else base_path
```

Comment:
Does this syntax work? I would imagine it should be `base_path + '/' + relpath`

Reply:
[Yes, it does! ](https://docs.python.org/3/library/pathlib.html#operators)